From 301cd3d88c60c4166450ae075864d6c325964468 Mon Sep 17 00:00:00 2001
From: Jaime Breva <jbreva@nayarsystems.com>
Date: Thu, 5 Jan 2023 03:12:59 +0800
Subject: [PATCH 2/3] arm64: dts: qcom: msm8916-zhihe-ufi001c: Enable modem

Enable modem for all zhihe-series devices.
GPIO 1 and 2 are used for enabling SIM detection
The reserved-memory section for mpss has to grow bigger, so gps_mem has
been relocated.

Co-developed-by: Yang Xiwen <forbidden405@foxmail.com>
Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>
---
 .../boot/dts/qcom/msm8916-zhihe-ufi001c.dts   |  4 ++++
 arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi   | 23 +++++++++++++++++--
 2 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts b/arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts
index df33b0be2ce1..5c364525a8c8 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts
@@ -33,3 +33,7 @@ &button_default {
 &gpio_leds_default {
 	pins = "gpio20", "gpio21", "gpio22";
 };
+
+&sim_ctrl_default {
+	pins = "gpio1", "gpio2";
+};
diff --git a/arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi b/arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi
index 85ac1a87e056..3f67ca3abf6e 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi
@@ -3,6 +3,7 @@
 /dts-v1/;
 
 #include "msm8916-pm8916.dtsi"
+#include "msm8916-modem.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
@@ -17,8 +18,13 @@ chosen {
 	};
 
 	reserved-memory {
-		gps_mem: gps@86800000 {
-			reg = <0x0 0x86800000 0x0 0x200000>;
+		mpss_mem: mpss@86800000 {
+			reg = <0x0 0x86800000 0x0 0x5500000>;
+			no-map;
+		};
+
+		gps_mem: gps@8bd00000 {
+			reg = <0x0 0x8bd00000 0x0 0x200000>;
 			no-map;
 		};
 	};
@@ -72,6 +78,11 @@ &gcc {
 	clocks = <&xo_board>, <&sleep_clk>, <0>, <0>, <0>, <0>, <0>;
 };
 
+&mpss {
+	pinctrl-0 = <&sim_ctrl_default>;
+	pinctrl-names = "default";
+};
+
 &pm8916_usbin {
 	status = "okay";
 };
@@ -216,4 +227,12 @@ gpio_leds_default: gpio-leds-default-state {
 		bias-disable;
 		drive-strength = <2>;
 	};
+
+	sim_ctrl_default: sim-ctrl-default-state {
+		function = "gpio";
+
+		bias-disable;
+		output-low;
+		drive-strength = <2>;
+	};
 };
-- 
2.38.1

